components/Map.js

```diff
@@ -0,0 +1,44 @@
+import { useState, useEffect } from "react";
+import ReactMapGL, { Marker } from "react-map-gl";
+export default function Map({ locations }) {
+  // UseState hook
+  const [selectedLocation, setSelectedLocation] = useState({});
+  const [viewport, setViewport] = useState({
+    width: "100%",
+    height: "100%",
+    // The latitude and longitude of the center of London
+    latitude: 40.0718,
+    longitude: -2.13401,
+    zoom: 12.5,
+  });
+  return (
+    <ReactMapGL
+      mapStyle="mapbox://styles/mapbox/streets-v11"
+      mapboxApiAccessToken="pk.eyJ1IjoiaGVjdG9ycGxpbmlvIiwiYSI6ImNrejQ3cDc2ZTBjbHEyb3J4MzMzZHpmMWMifQ.XT3g3xJFTaNGxYeMBtBoaQ"
+      {...viewport}
+      onViewportChange={(nextViewport) => setViewport(nextViewport)}
+    >
+      {locations.map((location) => (
```

Comment:
you are mapping an (I guess) array, but you are not having a safe operator on the function declaration. Knowing that you are not using Typescript you are not sure about what happen if someone doesn't add locations to the component. Which means you will have an undefined on locations variable. 

That means you can have an error here. So I encourage you to add safe operator on the definition

const Map = ({locations = []}) => {}

that safe operator gives you a default value (an empty array in this case) if the Map component doesn't have a location property.

Reply:
i understand what do you say but i don't know how do that, i will continue with anothers fails and after returned from this